### PR TITLE
Add event emitter option to track progress of computations

### PR DESCRIPTION
**Pull request by anyong**
If `callback` is not provided, the function returns an event emitter with event types `progress` and `done`.

Event descriptions:

```
const job = ccurlInterface(/* omit final callback argument */);

job.on('progress', (err, progress) => {
  // progress from 0 to 1
});

job.on('done', (err, result) => {
  // error or resulting trytes
});
```

### CODE REVIEW

**Comment by anyong** [comment]
@paulhandy @th0br0 Have either of you had a chance to review this?

**Comment by l3wi** [comment]
@th0br0 Can you please review this?

File: index.js [hunk 1]
```diff
@@ -1,45 +1,77 @@
...
+
+        if (callback) {
+            return callback(err, null);
+        } else {
+            emitter.emit('done', err, null);
+            return;
+        }
+    }
+
+    function reportProgress(count) {
```
Comment (th0br0) [inline index.js:27]: never called?

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (anyong) [reply index.js:27]: Fixed in f1651e4.

**Review by th0br0 — APPROVED** [review]

**Pull request merged**